contracts/SpokePool.sol

```diff
@@ -667,6 +675,60 @@ abstract contract SpokePool is
         _emitFillRelay(relayExecution, fillAmountPreFees);
     }
 
+    /**
+     * @notice Caller signals to the system that they want a refund on this chain, which they set as the
+     * `repaymentChainId` on the original fillRelay() call on the `destinationChainId`. An observer should be
+     * be able to 1-to-1 match the emitted RequestRefund event with the FilledRelay event on the `destinationChainId`.
+     * @dev This function could be used to artificially inflate the `fillCounter`, allowing the caller to "frontrun"
+     * and cancel pending fills in the mempool. This would in the worst case censor fills at the cost of the caller's
+     * gas costs. We don't view this as a major issue as the fill can be resubmitted and obtain the same incentive,
+     * since incentives are based on validated refunds and would ignore these censoring attempts. This is no
+     * different from calling `fillRelay` and setting msg.sender = recipient.
+     * @dev Caller needs to pass in `fillBlock` that the FilledRelay event was emitted on the `destinationChainId`.
+     * This is to prevent someone from requesting a refund before a corresponding fill has been mined.
+     * @param relayer Relayer on FilledRelay event on destinationChain we want to match with this Refund.
+     * This should be account receiving any refund stemming from this event.
+     * @param destinationToken This chain's destination token equivalent for original deposit destination token.
+     * @param amount Original deposit amount.
+     * @param originChainId Original origin chain ID.
+     * @param realizedLpFeePct Original realized LP fee %.
+     * @param depositId Original deposit ID.
+     * @param maxCount Max count to protect the refund recipient from frontrunning.
+     * @param fillBlock Block number of FilledRelay event on destinationChain we want to match with this Refund.
+     */
+    function requestRefund(
+        address relayer,
+        address destinationToken,
+        uint256 amount,
+        uint256 originChainId,
+        int64 realizedLpFeePct,
+        uint32 depositId,
+        uint256 maxCount,
+        uint256 fillBlock
```

Comment:
Do you think fillBlock actually helps our validation? I'm not convinced it does if you can put it indefinitely far into the future.

Ideally, we'd like a way to know if it's valid or invalid the moment it's submitted or block it from being submitted if it's referencing something in the future... do you think there's a value we could use for that? Maybe block timestamp? And require that the timestamp is strictly in the past? Or even require that the timestamp is > some configurable number of seconds before the current block?

Thoughts?

Reply:
Why would setting a fillBlock far into the future work? I was thinking `fillBlock` would have to be exactly correct to be valid, not as a lower limit

Reply:
Related to the duplicate counting, I think there will probably need to be rules around how the timing of the fill block relates to the timing of the refund request, but the duplicate counting ensures that you can submit twice safely with the understanding that the first must be deemed invalid for the second to be valid.

Reply:
@pxrl @james-a-morris @mrice32 as discussed off line there will be a rule added to the UMIP that the Refund event takes place no more than 20 minutes after the Fill event. This gives a margin of error for the destination and repayment chain to have timestamps that are off by 20 minutes (highly unlikely) and protects the dataworker a bit when deciding whether to validate a refund.